fn/slice.go

```diff
@@ -99,6 +99,18 @@ func Find[A any](pred func(A) bool, s []A) Option[A] {
 	return None[A]()
 }
 
+// FindIdx returns the first value that passes the supplied predicate along with
+// its index in the slice. If no satisfactory value is found, None is returned.
+func FindIdx[A any](pred func(A) bool, s []A) Option[T2[int, A]] {
```

Comment:
Stdlib has this now: https://pkg.go.dev/slices#Index

Could call it, then map the `-1` to an option.

Reply:
Turns out that it's not equivalent because [Index](https://pkg.go.dev/slices#Index) doesn't accept a predicate but rather checks for equality.